### PR TITLE
*: isolate more variables for `runWithSystemSession`

### LINKED ISSUE #54343: Warning 1356 is returned by querying information_schema.columns when a scalar subquery is used as a column in view definition
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)

```
mysql> create table t1(id int);
Query OK, 0 rows affected (0.06 sec)

mysql> create view v1 as select (select id from t1) as col from dual;
Query OK, 0 rows affected (0.07 sec)

mysql> select column_name, table_name from information_schema.columns where table_name='v1';
Empty set, 1 warning (0.00 sec)

mysql> show warnings;
+---------+------+---------------------------------------------------------------------------------------------------------------------------+
| Level   | Code | Message                                                                                                                   |
+---------+------+---------------------------------------------------------------------------------------------------------------------------+
| Warning | 1356 | View 'test.v1' references invalid table(s) or column(s) or function(s) or definer/invoker of view lack rights to use them |
+---------+------+---------------------------------------------------------------------------------------------------------------------------+
1 row in set (0.00 sec)
```

### 2. What did you expect to see? (Required)

```
+-------------+------------+
| column_name | table_name |
+-------------+------------+
| col         | v1         |
+-------------+------------+
```

### 3. What did you see instead (Required)

The warning 1356 is reported.

### 4. What is your TiDB version? (Required)

v6.1.0 and above

### PR DESCRIPTION
**Pull request by lance6716**
### What problem does this PR solve?

Issue Number: close #54343

Problem Summary:

### What changed and how does it work?

as title

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  > 

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: pkg/executor/executor.go [hunk 1]
```diff
@@ -1466,6 +1466,11 @@ func init() {
 			return nil, err
 		}
 
+		tm := sessiontxn.GetTxnManager(sctx)
+		err = tm.EnterNewTxn(ctx, &sessiontxn.EnterNewTxnRequest{})
```
Comment (D3Hunter) [inline pkg/executor/executor.go:1470]: should we commit or rollback before return?

**Review by tangenta — APPROVED** [review]

File: pkg/executor/infoschema_reader.go [hunk 2]
```diff
@@ -898,6 +897,7 @@ func (e *hugeMemTableRetriever) dataForColumnsInTable(ctx context.Context, sctx
 			internalCtx := kv.WithInternalSourceType(context.Background(), kv.InternalTxnOthers)
 			// Build plan is not thread safe, there will be concurrency on sessionctx.
 			if err := runWithSystemSession(internalCtx, sctx, func(s sessionctx.Context) error {
+				is := sessiontxn.GetTxnManager(s).GetTxnInfoSchema()
```
Comment (D3Hunter) [inline pkg/executor/infoschema_reader.go:900]: is this where the fix is, replace with another session ctx?

File: pkg/executor/show.go [hunk 3]
```diff
@@ -2377,5 +2378,12 @@ func runWithSystemSession(ctx context.Context, sctx sessionctx.Context, fn func(
 		return err
 	}
 	defer b.ReleaseSysSession(ctx, sysCtx)
+	if err = sessiontxn.NewTxn(ctx, sysCtx); err != nil {
```
Comment (D3Hunter) [inline pkg/executor/show.go:2381]: please add some comment on why we must open an explicit txn

File: pkg/executor/infoschema_reader.go (hunk 2 above)
Comment (lance6716) [reply pkg/executor/infoschema_reader.go:900]: The original error is "ctx provider not found", but it's overwritten https://github.com/pingcap/tidb/pull/54809. So the first fix is add `sessiontxn.NewTxn`.

Then another problem is the system session still uses the `is` captured by closure, which has conflict with user session. So moving it here.

**Review by D3Hunter — APPROVED** [review]

**Review by hawkingrei — APPROVED** [review]

**Pull request merged**